src/components/DatePicker/DatePicker.tsx

```diff
@@ -85,6 +85,16 @@ export class DatePicker extends React.Component<IDatePickerProps, IDatePickerSta
     this._validateTextInput = this._validateTextInput.bind(this);
   }
 
+  public componentWillReceiveProps(nextProps: IDatePickerProps) {
+    let { formatDate, value } = nextProps;
+
+    this.setState({
+      selectedDate: value || new Date(),
+      formattedDate: formatDate && value ? formatDate(value) : null,
+      errorMessage: ''
```

Comment:
do you really intend to clear the errorMessage whenever new props are passed in? That seems a little odd.